### PR TITLE
Train module refactor

### PR DESCRIPTION
**Pull request by roshankern**
This PR is ready for review!

In this PR, a model is created for each combination of:
```python
model_types = ["final", "shuffled_baseline"]
feature_types = ["CP", "DP", "CP_and_DP"]
```
these combinations are tested in a for loop to better reuse code. Each model is saved to the `models/` directory.

Also, the `1.split_data` module has a one-line change to include Greg's suggestion from https://github.com/WayScience/phenotypic_profiling_model/commit/2f50b408e3448251863eb5497b50cceeb66159c2#r103899488

### CODE REVIEW

File: 2.train_model/README.md [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
 
-In this module, we train a ML model to predict phenotypic class from DeepProfiler-extracted features.
+In this module, we train ML models to predict phenotypic class from cell features.
 
-We train the model in [train_model.ipynb](1.train_model.ipynb).
-We use [sklearn.linear_model.LogisticRegression](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html) for our machine learning model.
-We use the following parameters for our Logisic Regression model:
+We train the models in [train_model.ipynb](1.train_model.ipynb).
+We use [sklearn.linear_model.LogisticRegression](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html) for our machine learning models.
+A final model
```
Comment (gwaybio) [inline 2.train_model/README.md]: this looks like an incomplete sentence or typo?

File: 2.train_model/README.md [hunk 2]
```diff
@@ -22,9 +23,9 @@ We search over the following parameters: `[0. , 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.
 - `C`: Inversely proportional to regularization strength.
 We search over the following parameters: `[1.e-03, 1.e-02, 1.e-01, 1.e+00, 1.e+01, 1.e+02, 1.e+03]`
 
-The best parameters are used to train a final model on all of the training data.
-This final model is saved in [log_reg_model.joblib](models/log_reg_model.joblib).
-The model derived from shuffled training data is saved in [shuffled_baseline_log_reg_model.joblib](models/shuffled_baseline_log_reg_model.joblib).
+A final model is trained with CP features, DP features, and merged (CP and DP) features.
```
Comment (gwaybio) [inline 2.train_model/README.md]:
```suggestion
We train three final multiclass models with CP features, DP features, and merged (CP and DP) features.
```

Comment (gwaybio) [reply 2.train_model/README.md]: active voice please! :)

File: 2.train_model/README.md [hunk 3]
```diff
@@ -22,9 +23,9 @@ We search over the following parameters: `[0. , 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.
...
+A shuffled baseline model is trained for each of these feature types as well, with each column of the feature data being shuffled independently to create a shuffled baseline for comparison.
```
Comment (gwaybio) [inline 2.train_model/README.md]:
```suggestion
We also train shuffled baseline models for each of these feature types as well, with each column of the feature data being shuffled independently to create a shuffled baseline for comparison.
```

File: 2.train_model/README.md [hunk 4]
```diff
@@ -22,9 +23,9 @@ We search over the following parameters: `[0. , 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.
...
+Each model is saved in [models/](models/).
```
Comment (gwaybio) [inline 2.train_model/README.md]:
```suggestion
The notebooks save each model in [models/](models/).
```

**Review by gwaybio — APPROVED** [review]
Looks great 🤩 minor comments.

Also, the file name `final_CP__and_DP.joblib` looks a bit wonky (different than the other filenames). Specifically, the others have two underscores after the word final (`final__`) while this file name has two underscores after CP (`CP__`). Super minor but probably good to keep consistent.

File: 2.train_model/README.md (hunk 1 above)
Comment (roshankern) [reply 2.train_model/README.md]: A typo, fixed in https://github.com/WayScience/phenotypic_profiling_model/pull/20/commits/c2a67f2d3c048099a41dcf4ad606c374181af490

File: 2.train_model/README.md (hunk 2 above)
Comment (roshankern) [reply 2.train_model/README.md]: Ahhh! Of course

**Pull request merged**